client/src/main/java/io/pravega/client/netty/impl/ClientConnectionInboundHandler.java

```diff
@@ -66,7 +66,7 @@ public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
     }
 
     @Override
-    public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
```

Comment:
+1
`channelInactive ()` this is invoked when the channel registered is now inactive and reached its end of lifetime.
Also, `channelUnregistered()` is always invoked when `channelInactive()` is invoked.